libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -1542,69 +1598,32 @@ IotMqttError_t _IotMqtt_DeserializePublish( _mqttPacket_t * pPublish )
                 "DUP is 0." );
     }
 
-    /* Sanity checks for "Remaining length". */
-    if( pOutput->qos == IOT_MQTT_QOS_0 )
-    {
-        /* A QoS 0 PUBLISH must have a remaining length of at least 3 to accommodate
-         * topic name length (2 bytes) and topic name (at least 1 byte). */
-        if( pPublish->remainingLength < 3 )
-        {
-            IotLog( IOT_LOG_DEBUG,
-                    &_logHideAll,
-                    "QoS 0 PUBLISH cannot have a remaining length less than 3." );
+    /* Sanity checks for "Remaining length". A QoS 0 PUBLISH  must have a remaining
+     * length of at least 3 to accommodate topic name length (2 bytes) and topic
+     * name (at least 1 byte). A QoS 1 or 2 PUBLISH must have a remaining length of
+     * at least 5 for the packet identifier in addition to the topic name length and
+     * topic name. */
+    status = _checkRemainingLength( pPublish, pOutput->qos, 3, 5 );
```

Comment:
The comment is reasonably clear, but these constants should be #defines.

Reply:
Moved `3` to its own macro, and refactored to not need last argument.